gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-common/src/main/java/io/gravitee/am/gateway/handler/common/auth/user/impl/UserAuthenticationManagerImpl.java

```diff
@@ -256,7 +256,6 @@ private Maybe<UserAuthentication> loadUserByUsername0(Client client, Authenticat
                     Map<String, Object> additionalInformation = user.getAdditionalInformation() == null ? new HashMap<>() : new HashMap<>(user.getAdditionalInformation());
                     additionalInformation.put("source", authProvider);
                     additionalInformation.put(Parameters.CLIENT_ID, client.getId());
-                    ((DefaultUser) user).setAdditionalInformation(additionalInformation);
```

Comment:
removing this line may lead to regression as we test in line 256 if the additionalINformation is null. If so, we create a new HashMap but we do not set this map into the User object